java.lang.NumberFormatException: For input string: ""1688546411954""

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.ichi2.anki/com.ichi2.anki.NoteEditor}: java.lang.NumberFormatException: For input string: ""1688546411954""
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3834)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3980)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:101)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2394)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:211)
	at android.os.Looper.loop(Looper.java:300)
	at android.app.ActivityThread.main(ActivityThread.java:8294)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:559)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:954)
Caused by: java.lang.NumberFormatException: For input string: ""1688546411954""
	at java.lang.Long.parseLong(Long.java:722)
	at java.lang.Long.parseLong(Long.java:861)
	at com.ichi2.libanki.ConfigV16.getLong(SourceFile:12)
	at com.ichi2.libanki.Collection.get_config_long(SourceFile:11)
	at com.ichi2.anki.NoteEditor.setDid(SourceFile:81)
	at com.ichi2.anki.NoteEditor.onCollectionLoaded(SourceFile:333)
	at com.ichi2.anki.AnkiActivity.startLoadingCollection(SourceFile:28)
	at com.ichi2.anki.NoteEditor.onCreate(SourceFile:253)
	at android.app.Activity.performCreate(Activity.java:8522)
	at android.app.Activity.performCreate(Activity.java:8486)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1438)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3815)
	... 12 more
```
2.16.0beta5, new backend, Android 13
https://ankidroid.org/acra/app/1/bug/57860/report/26d7efca-dc5e-4707-b740-0973589373bb

--- a/AnkiDroid/src/test/java/com/ichi2/anki/NoteEditorTest.kt
+++ b/AnkiDroid/src/test/java/com/ichi2/anki/NoteEditorTest.kt
@@ -25,6 +25,7 @@ import androidx.test.ext.junit.runners.AndroidJUnit4
 import com.ichi2.anki.AbstractFlashcardViewer.Companion.editorCard
 import com.ichi2.anki.NoteEditorTest.FromScreen.DECK_LIST
 import com.ichi2.anki.NoteEditorTest.FromScreen.REVIEWER
+import com.ichi2.anki.api.AddContentApi.Companion.DEFAULT_DECK_ID
 import com.ichi2.anki.multimediacard.activity.MultimediaEditFieldActivity
 import com.ichi2.libanki.Consts
 import com.ichi2.libanki.Decks.Companion.CURRENT_DECK
@@ -347,6 +348,14 @@ class NoteEditorTest : RobolectricTest() {
         assertEquals("012345text", field.fieldText)
         assertEquals(10, field.selectionStart)
         assertEquals(10, field.selectionEnd)
+    }
+
+    @Test
+    fun `can open with corrupt current deck - Issue 14096`() {
+        col.config.set(CURRENT_DECK, '"' + "1688546411954" + '"')
+        getNoteEditorAddingNote(DECK_LIST, NoteEditor::class.java).apply {
+            assertThat("current deck is default after corruption", deckId, equalTo(DEFAULT_DECK_ID))
+        }
     }
 
     private fun getCopyNoteIntent(editor: NoteEditor): Intent {

--- a/AnkiDroid/src/test/java/com/ichi2/libanki/ConfigTest.kt
+++ b/AnkiDroid/src/test/java/com/ichi2/libanki/ConfigTest.kt
@@ -71,6 +71,17 @@ class ConfigTest : JvmTest() {
             col.config.set("map2", map2)
         }
     }
+
+    @Test
+    fun `A string value is handled as a long - Issue 14096`() {
+        col.config.set("test", '"' + "1688546411954" + '"')
+        col.config.set("test1", "1688546411954")
+        col.config.set("test2", 1688546411954)
+
+        assertThat(col.config.get<Long>("test"), equalTo(null))
+        assertThat(col.config.get<Long>("test1"), equalTo(1688546411954L))
+        assertThat(col.config.get<Long>("test2"), equalTo(1688546411954L))
+    }
 }
 
 @Serializable

